### DISCUSSION TITLE
Sane Image Size - Auto Reduction

### DISCUSSION THREAD

**Discussion by NilsBaumgartner1994** [directus/directus#22622] [Feature Requests]
### Summary

Automatically scale down large images or files to a sane size.
Idea based on hook: https://github.com/martabitbrain/directus-extension-sane-image-size

### Motivation

Sometimes users dont look what large files or images they upload. would be nice not only to transform the image, but to reduce the disk space in first place.

### Detailed Design

Use sharp to reduce the image size to a specific width and height or megabyte size.

### Must Have:
-  Variable to enable auto resize

### Should Have:
- Optinal parameters to set them up
- Enabled by default

### Could Have:
- Linking to existing transformator

### Won't Have:
- 

### Drawbacks

-

### Alternatives

Keeping the hook solution, which might break in the future

### Adoption Strategy

No breaking changes.

**Comment by Resumane** [comment]
Agreed, and since it's related, I would add that the ability to add a watermark on uploaded pictures would be very convenient.
Having to compress the pictures, resize them and watermark them before we upload them to Directus feels like a wasted opportunity, these common sense features are available on other CMSs.

**Reply by NilsBaumgartner1994** [reply]
Oh that watermark is very good point.

**Reply by kolosseo** [reply]
> Agreed, and since it's related, I would add that the ability to add a watermark on uploaded pictures would be very convenient. Having to compress the pictures, resize them and watermark them before we upload them to Directus feels like a wasted opportunity, these common sense features are available on other CMSs.

To add a watermark you can use the native Directus features.
Upload a semi-transparent image to be used as a watermark. Once uploaded, copy the image ID.
Go to:
  -> Settings
  -> Transformation Presets
  -> Create a "Transformation Preset" (you can call it, for example `my-watermark`)
  -> Additional Transformations
  -> add this code:
```
[
    [
        "composite",
        [
            {
                "blend": "screen",
                "input": "/directus/uploads/ID_OF_YOUR_WATERMARK_IMAGE"
            }
        ]
    ]
]
```

Now, to see the watermarked images you have to add the suffix `?key=PRESET_NAME` to the CMS images link.
Example: `https://mywebsite.com/assets/IMAGE_ID?key=my-watermark`.
